[CI] AsyncSearchErrorTraceIT testDataNodeDoesNotLogStackTraceWhenErrorTraceTrue failing

**Build Scans:**
- [elasticsearch-periodic #7550 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/uarljwsr63lo6)
- [elasticsearch-periodic #7515 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/wxxn2ov6h57hu)
- [elasticsearch-periodic-platform-support #7516 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/62m3c6u7tzyz6)
- [elasticsearch-periodic-platform-support #7474 / debian-11_platform-support-unix](https://gradle-enterprise.elastic.co/s/yulytljfwnzsg)
- [elasticsearch-periodic-platform-support #7467 / windows-2019_checkpart2_platform-support-windows](https://gradle-enterprise.elastic.co/s/j5qxrtj64csys)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:async-search:internalClusterTest" --tests "org.elasticsearch.xpack.search.AsyncSearchErrorTraceIT.testDataNodeDoesNotLogStackTraceWhenErrorTraceTrue" -Dtests.seed=8FE42AAB71B1EDC -Dtests.locale=nus-SS -Dtests.timezone=Brazil/DeNoronha -Druntime.java=24
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.search.AsyncSearchErrorTraceIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testDataNodeDoesNotLogStackTraceWhenErrorTraceTrue),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Shard [.async-search][0] is still locked after 5 sec waiting
```

**Issue Reasons:**
- [8.x] 5 failures in test testDataNodeDoesNotLogStackTraceWhenErrorTraceTrue (1.0% fail rate in 518 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (13.3% fail rate in 15 executions)
- [8.x] 3 failures in pipeline elasticsearch-periodic-platform-support (18.8% fail rate in 16 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -417,9 +417,6 @@ tests:
 - class: org.elasticsearch.xpack.downsample.ILMDownsampleDisruptionIT
   method: testILMDownsampleRollingRestart
   issue: https://github.com/elastic/elasticsearch/issues/126495
-- class: org.elasticsearch.xpack.search.AsyncSearchErrorTraceIT
-  method: testDataNodeLogsStackTraceWhenErrorTraceFalseOrEmpty
-  issue: https://github.com/elastic/elasticsearch/issues/126357
 - class: org.elasticsearch.smoketest.MlWithSecurityIT
   method: test {yaml=ml/get_trained_model_stats/Test get stats given trained models}
   issue: https://github.com/elastic/elasticsearch/issues/126510

--- a/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchErrorTraceIT.java
+++ b/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchErrorTraceIT.java
@@ -185,7 +185,6 @@ public class AsyncSearchErrorTraceIT extends ESIntegTestCase {
                 responseEntity = performRequestAndGetResponseEntityAfterDelay(request, TimeValue.timeValueSeconds(1L));
             }
 
-            getRestClient().performRequest(searchRequest);
             mockLog.assertAllExpectationsMatched();
         }
     }
@@ -228,7 +227,6 @@ public class AsyncSearchErrorTraceIT extends ESIntegTestCase {
                 responseEntity = performRequestAndGetResponseEntityAfterDelay(request, TimeValue.timeValueSeconds(1L));
             }
 
-            getRestClient().performRequest(searchRequest);
             mockLog.assertAllExpectationsMatched();
         }
     }

